account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -246,6 +261,13 @@ def _pull(self, date_since, date_until):
                                 )
                             )
                         continue
+
+                    if isinstance(date, datetime):
```

Comment:
it is the wrong place to do convert. Because you convert dateime UTC to 
you sould do that here : 
https://github.com/OCA/bank-statement-import/blob/12.0/account_bank_statement_import_online/models/online_bank_statement_provider.py#L342
create an other to convert line_values['date'] in date:
```python
def _convert_datetime_in_date(self, date):
     return date.astimezone(self).date()
```
`line_values['date'] = self._convert_datetime_in_date(date)`

update this methode to have the good date in the statement.
```python
    def _get_statement_date(self, date_since, date_until):
        self.ensure_one()
        return date_since.astimezone(provider_tz).date()
```

no ?

I think for better reading of the code, `date_since` and `date_until` could be a `date` and not `datetime`, and `_obtain_statement_data` could be return line with `date` (and not `datetime`).

Reply:
Thanks, regarding statement date it is a valid point - added. As for converting to date prior to returning data - we could do that, yet it's done automatically by Odoo, and just in case there would be a module that changes date field to be datetime, this would make it compatible.